src/games/strategy/engine/framework/map/download/InstallMapDialog.java

```diff
@@ -53,37 +56,65 @@ public static Version getVersion(final File zipFile) {
     return props.getVersion();
   }
 
+  /**
+   * Shows the download window and begins downloading the specified map right away.
+   * If the map cannot be downloaded a message prompt is shown to the user.
+   */
+  public static void showDownloadMapsWindow(final String mapName) {
+    showDownloadMapsWindow(null, Optional.of(mapName));
+  }
+
+  public static void showDownloadMapsWindow(final Component parent) {
+    showDownloadMapsWindow(parent, Optional.empty());
+  }
+
+  public static void showDownloadMapsWindow() {
+    showDownloadMapsWindow(null, Optional.empty());
+  }
+
 
-  public static void showDownloadMapsWindow(final Component parent, final List<DownloadFileDescription> games) {
+  private static void showDownloadMapsWindow(final Component parent, Optional<String> mapName) {
```

Comment:
This line is perhaps the key to the change. When we create a 'download maps window' we give an optional map name which is a map that we wish to start downloading immediately.